views/components/ConfigurationsList/index.jsx

```diff
@@ -0,0 +1,851 @@
+
+import { fetchUsernames } from '../../fe-utils/fetchUtil'
+import openNewWindow from '../../fe-utils/windowUtil'
+
+const basePath = basePathConfig || ''
+
+function NoOptionsMessage(props) {
+  return (
+    <Typography
+      color="textSecondary"
+      className={props.selectProps.classes.noOptionsMessage}
+      {...props.innerProps}
+    >
+      {props.children}
+    </Typography>
+  )
+}
+
+function inputComponent({ inputRef, ...props }) {
+  return <div ref={inputRef} {...props} />
+}
+
+function Control(props) {
+  return (
+    <TextField
+      fullWidth
+      InputProps={{
+        inputComponent,
+        inputProps: {
+          className: props.selectProps.classes.input,
+          inputRef: props.innerRef,
+          children: props.children,
+          ...props.innerProps,
+        },
+      }}
+      {...props.selectProps.textFieldProps}
+    />
+  )
+}
+
+function Option(props) {
+  return (
+    <MenuItem
+      buttonRef={props.innerRef}
+      selected={props.isFocused}
+      component="div"
+      style={{
+        fontWeight: props.isSelected ? 500 : 400,
+      }}
+      {...props.innerProps}
+    >
+      {props.children}
+    </MenuItem>
+  )
+}
+
+function Placeholder(props) {
+  return (
+    <Typography
+      color="textSecondary"
+      className={props.selectProps.classes.placeholder}
+      {...props.innerProps}
+    >
+      {props.children}
+    </Typography>
+  )
+}
+
+function SingleValue(props) {
+  return (
+    <Typography
+      className={props.selectProps.classes.singleValue}
+      {...props.innerProps}
+    >
+      {props.children}
+    </Typography>
+  )
+}
+
+function ValueContainer(props) {
+  return (
+    <div className={props.selectProps.classes.valueContainer}>
+      {props.children}
+    </div>
+  )
+}
+
+function MultiValue(props) {
+  return (
+    <Chip
+      tabIndex={-1}
+      label={props.children}
+      className={classNames(props.selectProps.classes.chip, {
+        [props.selectProps.classes.chipFocused]: props.isFocused,
+      })}
+      onDelete={props.removeProps.onClick}
+      deleteIcon={<CancelIcon {...props.removeProps} />}
+    />
+  )
+}
+
+function Menu(props) {
+  return (
+    <Paper
+      square
+      className={props.selectProps.classes.paper}
+      {...props.innerProps}
+    >
+      <MenuList>{props.children}</MenuList>
+    </Paper>
+  )
+}
+
+const components = {
+  Control,
+  Menu,
+  MultiValue,
+  NoOptionsMessage,
+  Option,
+  Placeholder,
+  SingleValue,
+  ValueContainer,
+}
+const ConfigurationsList = ({ user, classes, theme }) => {
+  const [state, setState] = useState({
+    user: {},
+    preferences: {},
+    configurations: [],
+    gridCols: 1,
+    gridWidth: 350,
+    searchUsers: false,
+    friends: [],
+    shared: [],
+    snackTime: false,
+    uploadSnack: false,
+    selectedConfig: {},
+    configOwner: '',
+    totalStudies: 0,
+    totalSubjects: 0,
+    totalDays: 0,
+  })
```

Comment:
should we split these up into individual state attributes? That might also highlight the attributes we don't need.

Reply:
#296 is scoped for this